### ISSUE TITLE
pragma region attach bug

### ISSUE THREAD

**Issue by ecossevin** [ecmwf-ifs/loki#427]
### What happened?

When I do with pragma region attach with NAME=CPTEND in the pragma, the pragma region isn't created.

### What are the steps to reproduce the bug?

SUBROUTINE TEST()
IMPLICIT NONE
!$loki parallel PARALLEL, TARGET=OpenMP/OpenMPSingleColumn/OpenACCSingleColumn,NAME=CPTEND_NEW 
!$loki end parallel 
!$loki parallel PARALLEL, TARGET=OpenMP/OpenMPSingleColumn/OpenACCSingleColumn,NAME=CPTEND 
!$loki end parallel
!$loki parallel PARALLEL, TARGET=OpenMP/OpenMPSingleColumn/OpenACCSingleColumn,NAME=ACNEBCOND
!$loki end parallel
!$loki parallel PARALLEL, TARGET=OpenMP/OpenMPSingleColumn/OpenACCSingleColumn,NAME=CPTEND
!$loki end parallel
!$loki parallel PARALLEL, TARGET=OpenMP/OpenMPSingleColumn/OpenACCSingleColumn,NAME=TOTO
!$loki end parallel
END SUBROUTINE TEST

from loki import *
src_file = "sub.F90"
src_name = "TEST"
source = Sourcefile.from_file(src_file)
routine = source[src_name]

with pragma_regions_attached(routine):
    print(routine.body.body)
    breakpoint()

### Version

0.2.7 Latest

### Platform (OS and architecture)

Linux pxdesrtls020 6.8.0-47-generic #47~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Oct  2 16:16:55 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

### Relevant log output

```shell
(loki_env) (base) [cossevine@taranislogin0 pragma 08:50:39]$ python3 test.py 
[Loki::Sourcefile] Constructed from sub.F90 in 0.03s
(Pragma:: loki parallel PARA..., Pragma:: loki end parallel, Pragma:: loki parallel PARA..., Pragma:: loki end parallel, PragmaRegion::, Pragma:: loki parallel PARA..., Pragma:: loki end parallel, PragmaRegion::)
> /home/gmap/mrpm/cossevine/bug_loki/pragma/test.py(7)<module>()
-> with pragma_regions_attached(routine):


Pragma:: loki parallel PARA..., Pragma:: loki end parallel, SHOULD BE PragmaRegion::
```

**Comment by mlange05** [comment]
Hi @ecossevin , many thanks for this. This looks a lot like a problem I've recently fixes in this commit:
https://github.com/ecmwf-ifs/loki/commit/2f9b51d6a845baa137dbf640a16a82ae645a6677

Have you confirmed this is still a problem with either `v0.2.7` or latest `main`?

**Comment by ecossevin** [comment]
Yes I did a pip install of loki to have the newest version. Is there a command to check the version of loki ?

**Comment by mlange05** [comment]
Hi, can you please try what you get for `python -c "import loki; print(loki.__version__)"`?

Since loki is still distributed via git checkouts, you might also have to do something like
```
cd <loki>
git checkout v0.2.7
pip install -e .
python -c "import loki; print(loki.__version__)"
```

**Comment by ecossevin** [comment]
```
(loki) (base) [cossevine@taranislogin0 loki 10:54:42]$ python -c "import loki; print(loki.__version__)"
0.2.8.dev182+gb833fef1
```

**Comment by mlange05** [comment]
Hi @ecossevin , apologies, I've just looked at this in a little more detail, and I had already fixed this in one of my dev branches, but hadn't submitted the fix yet. I've just filed the fix as PR #431. Could you please confirm that this fixes your issue?
